libraries/SuplaDevice/SuplaDevice.cpp

```diff
@@ -305,7 +305,8 @@ bool SuplaDeviceClass::begin(IPAddress *local_ip, char GUID[SUPLA_GUID_SIZE], ui
 	srpc_params.user_params = this;
 	
 	srpc = srpc_init(&srpc_params);
-	
+    srpc_set_proto_version(srpc, 10); // Set Supla protocol interface version to 10
```

Comment:
This should not be hard-coded.
Maybe as a parameter of the SuplaDevice constructor?
There would be two. One for setting the version and the other for the default value.

Reply:
One constructor will be enough. Currently SuplaDevice is created in SuplaDevice.cpp file in library. This is not a nice place to allow user edits. Maybe in begin()? But then all examples will have to be modified to set version 10 before 11 goes into production.

Reply:
Why 7? According to proto.h, THERMOMETER, HUMIDITY, PRESSURE etc sensors require 8. I also plan to merge impulse counter soon (v 10).

Reply:
The most private instances we monitor are still with 7 but you can go to 8.
You should print warnings that the channel will not work with the specified version. Therefore, the version in the constructor would be better. Then you can print a warning at the level of adding channels. Otherwise, warnings should be printed in the begin () method. Regardless of the warnings, srpc will check whether a particular call is possible. 

https://github.com/SUPLA/supla-core/blob/master/supla-common/srpc.c#L1159

Reply:
Maybe I'll add some version negotiation? I.e. register with highest version from proto.h and then in case of "protocol version error" go down to highest supported version by server?

Calls will be verified and blocked by srpc. How server will react to channels of unknown type?

For now I'll do quick fix and set default to 8 with option to modify in begin(). Later I'll add version verification on each channel type.